src/compiler/sys.ts

```diff
@@ -298,6 +298,55 @@ namespace ts {
                 };
             }
 
+            function createWatchedFileSet() {
+                const watchedDirectories = createFileMap<FileWatcher>();
+                const watchedFiles = createFileMap<(fileName: string, removed?: boolean) => void>();
+                const currentDirectory = process.cwd();
+
+                return { addFile, removeFile };
+
+                function addFile(fileName: string, callback: (fileName: string, removed?: boolean) => void): WatchedFile {
+                    const path = toPath(fileName, currentDirectory, getCanonicalPath);
+                    const parentDirPath = toPath(ts.getDirectoryPath(fileName), currentDirectory, getCanonicalPath);
+
+                    if (!watchedDirectories.contains(parentDirPath)) {
+                        watchedDirectories.set(parentDirPath, _fs.watch(
+                            parentDirPath,

```

Comment:
The Node docs state "For example, watching files or directories on network file systems (NFS, SMB, etc.) often doesn't work reliably or at all. You can still use fs.watchFile, which uses stat polling, but it is slower and less reliable.". 

Do we detect this and fall back? (Is it even possible to?) If not, how will users know if they get in this state and get things working correctly? I believe we've definitely seen users log issues where they are opening projects off the network.